lib/awful/rules.lua

```diff
@@ -307,6 +308,32 @@ function rules.extra_properties.geometry(c, _, props)
     c:geometry(new_geo) --TODO use request::geometry
 end
 
+--- Create a new tag based on a rule.
+-- @tparam client c The client
+-- @tparam boolean|function|string value The value.
+-- @treturn tag The new tag
+function rules.high_priority_properties.new_tag(c, value)
+    local ty = type(value)
+    local t = nil
+
+    if ty == "boolean" then
+        -- Create a new tag named after the client class
+        t = atag.add(c.class or "N/A", {screen=c.screen})
+    elseif ty == "string" then
+        -- Create a tag named after "value"
+        t = atag.add(value, {screen=c.screen})
```

Comment:
Shouldn't both this and the previous case also use `volatile=true` for the new tag?